### PR TITLE
Enable performance test API -- Latency

### PR DESCRIPTION
**Pull request by yuzhyang**
Add performance test api for CLI image. This function support latency test currently in ot-cli-ftd of nrf52840 platform which is disabled by default.

### CODE REVIEW

File: src/cli/cli_latency.cpp [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+
+// #include <openthread/config.h>
+#include <openthread/message.h>
+#include <openthread/udp.h>
+
+#include "cli/cli.hpp"
+#include "cli_uart.hpp"
+#include "common/encoding.hpp"
+
+#define MONITOR_PIN 3
```
Comment (Nopik) [inline src/cli/cli_latency.cpp]: gpio.c has apparently this magic pin number in 2 different places, this is third one. Compiling this on custom board will require changing bunch of `3`s to different pin. Can we get one `#define` for it?

**Comment by chshu** [comment]
The "pretty-check" failed, please run following cmd locally to format the changes:
```
./bootstrap
./configure
make pretty
```

File: src/cli/cli_latency.cpp [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+#define CLEAR_PIN_INTERVAL 2
```
Comment (librasungirl) [inline src/cli/cli_latency.cpp]: add comment for the short interval to clear pin?

File: src/cli/cli_latency.cpp [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+        if (sSendTimestamp[i] == 0)
+            break;
+        mInterpreter.mServer->OutputFormat("%d, %d, %d \r\n", ot::Cli::CliLatency::sSendTimestamp[i], mReceiveTimer[i], i);
+    }
+
+    return OT_ERROR_NONE;
+} 
+
+
+otError CliLatency::ProcessGpio(int argc, char *argv[])
```
Comment (librasungirl) [inline src/cli/cli_latency.cpp]: what's the purpose for this method?

File: src/cli/cli_latency.cpp [hunk 4]
```diff
@@ -0,0 +1,557 @@
...
+
+const struct CliLatency::Command CliLatency::sCommands[] =
+{
+    { "help", &CliLatency::ProcessHelp },
+    { "bind", &CliLatency::ProcessBind },
+    { "close", &CliLatency::ProcessClose },
+    { "connect", &CliLatency::ProcessConnect },
+    { "open",  &CliLatency::ProcessOpen },
+    { "start", &CliLatency::ProcessStart },
+    { "send", &CliLatency::ProcessSend },
```
Comment (librasungirl) [inline src/cli/cli_latency.cpp]: what's the purpose for `send` cmd? Below `test` cmd should be enough for test?

File: examples/platforms/nrf52840/gpio.c [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements the OpenThread platform GPIO driver on nrf52840.
+ *
+ */
+#include <openthread/config.h>
+#undef PACKAGE
```
Comment (librasungirl) [inline examples/platforms/nrf52840/gpio.c]: remove this line?

File: examples/platforms/nrf52840/gpio.c [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+#include <openthread/platform/logging.h>
+#include <utils/code_utils.h>
+
+#include <drivers/clock/nrf_drv_clock.h>
+#include <hal/nrf_uart.h>
+#include <hal/nrf_gpio.h>
+#include <hal/nrf_gpiote.h>
+#include "platform-nrf5.h"
+
+#define IN_PIN 3
```
Comment (librasungirl) [inline examples/platforms/nrf52840/gpio.c]: not used. remove?

File: examples/platforms/nrf52840/gpio.c [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+}
+
+void nrf5GpioProcess(otInstance *aInstance)
+{
+    (void) aInstance;
+    if (nrf_gpio_pin_read(3) > 0)
+    {
+        if (state != 1)
+        {
+            otLogCritPlat(sInstance, " %d", state);
```
Comment (librasungirl) [inline examples/platforms/nrf52840/gpio.c]: use parameter `aInstance` and may be not necessary to keep the variable `sInstance`?

File: examples/platforms/nrf52840/gpio.c [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+         * in ascending order. */
+        event = (nrf_gpiote_events_t)((uint32_t)event + sizeof(uint32_t));
+    }
+
+    otPlatGpioSignalEvent(sPin);
+}
+
+void GPIOTE_IRQHandler(void)
+{
+    nrf_drv_gpiote_evt_handler_t(sPin, GPIOTE_CONFIG_POLARITY_LoToHi);
```
Comment (librasungirl) [inline examples/platforms/nrf52840/gpio.c]: what if multiple Pins' interrupts? how to distinguish?

**Comment by jwhui** [comment]
Closing stale PR.

**Pull request closed without merging**